keygen/src/keygen.rs

```diff
@@ -310,78 +374,167 @@ fn main() -> Result<(), Box<dyn error::Error>> {
+            for el in starting_messages {
+                println!("\t{}",el);
+            }
+
+            let grind_matches_thread_safe = Arc::new(grind_matches);
             let attempts = Arc::new(AtomicU64::new(1));
             let found = Arc::new(AtomicU64::new(0));
             let start = Instant::now();
 
-            println!(
-                "Searching with {} threads for a pubkey containing {:?} or starting with {:?}",
-                num_cpus::get(),
-                includes,
-                starts_with
-            );
-
             let _threads = (0..num_cpus::get())
                 .map(|_| {
                     let attempts = attempts.clone();
                     let found = found.clone();
-                    let includes = includes.clone();
-                    let starts_with = starts_with.clone();
+                    let grind_matches_thread_safe = grind_matches_thread_safe.clone();
 
                     thread::spawn(move || loop {
                         let attempts = attempts.fetch_add(1, Ordering::Relaxed);
-                        if attempts % 5_000_000 == 0 {
+                        if attempts % 1_000_000 == 0 {
                             println!(
-                                "Searched {} keypairs in {}s. {} matches found",
+                                "Searched {} keypairs in {}s. {} matches found.",
                                 attempts,
                                 start.elapsed().as_secs(),
                                 found.load(Ordering::Relaxed),
                             );
                         }
-
                         let keypair = Keypair::new();
                         let mut pubkey = bs58::encode(keypair.pubkey()).into_string();
-
                         if ignore_case {
                             pubkey = pubkey.to_lowercase();
                         }
-
-                        if starts_with.iter().any(|s| pubkey.starts_with(s))
-                            || includes.iter().any(|s| pubkey.contains(s))
-                        {
-                            let found = found.fetch_add(1, Ordering::Relaxed);
-                            output_keypair(
-                                &keypair,
-                                &format!("{}.json", keypair.pubkey()),
-                                &format!("{}", found),
-                            )
-                            .unwrap();
+                        let mut total_matches_found = 0;
+                        for i in 0..grind_matches_thread_safe.len() {
+                            if grind_matches_thread_safe[i].count == grind_matches_thread_safe[i].found.load(Ordering::Relaxed) {
+                                total_matches_found = total_matches_found + 1;
+                                continue;
+                            }
+                            if !grind_matches_thread_safe[i].starts.is_empty() && 
```

Comment:
It would be nice to condense these three if blocks into a single block.

Reply:
updated